alfresco-health-processor-platform/src/main/java/eu/xenit/alfresco/healthprocessor/plugins/solr/SolrIndexValidationHealthProcessorPlugin.java

```diff
@@ -87,20 +85,24 @@ protected Logger getLogger() {
                         searchResult);
 
```

Comment:
> The sharding mechanism (communication) is not tested. (No request with the sharding parameter)

What sharding parameter are you referring to? The one used for normal Alfresco searches? We don't use that for index checks because it leaves us with no control over which nodes the search will actually be sent to.

We iterate over all shards that *should* contain the node, so that also covers sharded setups for which we have some code to determine which shards should contain the node.

> In case of non DB_ID_RANGE, duplicates are not detected.

Yes, there is a case where duplicates are not detected: when nodes are moved across shards, we don't check the shard where the node should have been removed from.
That is also true in some sense for DB_ID_RANGE shards, if someone decides to change shard ranges on the fly without doing a full re-index of the node.

Since duplicates are:

1. very rare to occur
2. not as problematic as nodes missing from the index (people can see and disregard a duplicate result, they can't see a missing document)

I think it is okay to not send search requests to all shards, just to verify that there are certainly no duplicates.

I will add these remarks to the duplicate node detection documentation as well. We can always revisit and change our logic later on, if we find that duplicates are a big problem.